tmt/result.py

```diff
@@ -299,14 +299,39 @@ def failures(log: Optional[str], msg_type: str = 'FAIL') -> str:
             return ''
         filtered = ''
 
-        # Filter beakerlib style logs, reverse the log string by lines, search for each FAIL
-        # and every associated line, then reverse the picked lines back into correct order
-        for m in re.findall(
-                fr'(^.*\[\s*{msg_type}\s*\][\S\s]*?)(?:^::\s+\[[0-9: ]+|:{{80}})',
-                '\n'.join(log.split('\n')[::-1]), re.MULTILINE):
-            filtered += m.strip() + '\n'
-        if filtered:
-            return '\n'.join(filtered.strip().split('\n')[::-1])
+        # Filter beakerlib style logs in the following way:
+        # 1. Reverse the log string by lines
+        # 2. Search for each FAIL and extract every associated line.
+        # 3. For failed phases also extract phase name so the log is easier to understand
+        # 4. Reverse extracted lines back into correct order.
+        if re.search(':: \\[   FAIL   \\] ::', log):  # dumb check for a beakerlib log
```

Comment:
I would rather check `test_framework` directly but I do not know how to access it from here.

Reply:
Sorry, there's no link to the original `base.Test` instance. We may add it, but that would be a much bigger patch.

One worth doing in the future, though - when I'm reading this, I think this belongs to the test framework, into some kind of `failures()` method. This is all fairly beakerlib-ish, I'd put it into `tmt.frameworks.beakerlib` then. Can't do so yet, not without the link to test and it framework.

Reply:
Exactly, I thought the same, such a code belongs to framework itself.

Reply:
I filed https://github.com/teemtee/tmt/issues/2355 to track the refactoring, it's the best I can offer right now.

Reply:
Yeah, makes sense.